### PR TITLE
Generate non csf certificates

### PR DESCRIPTION
**Pull request by Hamms**
Note that the design of the core image - the way the title wraps around the name field - means that particularly long names will overlap the title: 

image

However, also note that a quick look at the data confirms that we rarely need to deal with names over 20 characters
teacher hist

And names in that range fit just fine:

image

Note also that the facilitator names are implemented by stacking them, meaning that with a particularly large number of facilitators we could run into similar problems, but it would take quite a few for us to get to that point, and three facilitators (the maximum expected number) fit quite comfortably:

image

### CODE REVIEW

**Comment by mehalshah** [comment]
tagging @tanyaparker to test something

File: dashboard/app/controllers/pd/csf_certificate_controller.rb [hunk 1]
```diff
@@ -11,5 +11,7 @@ def generate_certificate
     )
 
     send_data image.to_blob, type: 'image/png', disposition: 'inline'
+  ensure
+    image && image.destroy!
```
Comment (mehalshah) [inline dashboard/app/controllers/pd/csf_certificate_controller.rb]: image.try(:destroy!) ?

File: dashboard/app/controllers/pd/workshop_certificate_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+class Pd::WorkshopCertificateController < ApplicationController
+  before_action :authenticate_user!
+  load_resource :enrollment, class: 'Pd::Enrollment', find_by: :code, id_param: :enrollment_code
+
+  def generate_certificate
+    workshop = @enrollment.workshop
+
+    facilitator_names = workshop.facilitators.map {|f| f.name.strip}
+    facilitator_fields = facilitator_names.each_with_index.map do |name, i|
```
Comment (mehalshah) [inline dashboard/app/controllers/pd/workshop_certificate_controller.rb]: Is there any ordering of facilitator names we should do? Ask the PLC team this

File: dashboard/app/controllers/pd/csf_certificate_controller.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/app/controllers/pd/csf_certificate_controller.rb]: why?

**Comment by Hamms** [comment]
PTAL.

Note that I cherry-picked a commit from [this pending PR](https://github.com/code-dot-org/code-dot-org/pull/13894) to help with the eyes test

**Comment by aoby** [comment]
Reverting because the new eyes test is failing with
```
Mysql2::Error: Field 'pd_workshop_id' doesn't have a default value: INSERT INTO `pd_enrollments` (`first_name`, `last_name`, `email`, `created_at`, `updated_at`, `code`, `school_info_id`) VALUES ('TestTeacher9681c657a0', 'Last - 11881f3a76b76d9aec1ce1639d281398', 'user@example.com', '2017-03-29 21:21:44', '2017-03-29 21:21:44', '9c3d780510496b97eaf2', 2) (ActiveRecord::StatementInvalid)
./step_definitions/pd.rb:89:in `create_enrollment'
./step_definitions/pd.rb:162:in `/^I create a workshop for course "([^"]*)" ([a-z]+) by "([^"]*)" with (\d+) (people|facilitators)(.*)$/'
./step_definitions/pd.rb:42:in `/^I am a teacher who has just followed a workshop certificate link$/'
./support/hooks.rb:36:in `call'
./support/hooks.rb:36:in `block in '
features/pd/workshopCertificates.feature:7:in `Given I am a teacher who has just followed a workshop certificate link'
./step_definitions/pd.rb:89:in `create_enrollment'
./step_definitions/pd.rb:162:in `/^I create a workshop for course "([^"]*)" ([a-z]+) by "([^"]*)" with (\d+) (people|facilitators)(.*)$/'
./step_definitions/pd.rb:42:in `/^I am a teacher who has just followed a workshop certificate link$/'
./support/hooks.rb:36:in `call'
./support/hooks.rb:36:in `block in '
features/pd/workshopCertificates.feature:7:in `Given I am a teacher who has just followed a workshop certificate link'
./step_definitions/pd.rb:89:in `create_enrollment'
./step_definitions/pd.rb:162:in `/^I create a workshop for course "([^"]*)" ([a-z]+) by "([^"]*)" with (\d+) (people|facilitators)(.*)$/'
./step_definitions/pd.rb:42:in `/^I am a teacher who has just followed a workshop certificate link$/'
./support/hooks.rb:36:in `call'
./support/hooks.rb:36:in `block in '
features/pd/workshopCertificates.feature:7:in `Given I am a teacher who has just followed a workshop certificate link'
./step_definitions/pd.rb:89:in `create_enrollment'
./step_definitions/pd.rb:162:in `/^I create a workshop for course "([^"]*)" ([a-z]+) by "([^"]*)" with (\d+) (people|facilitators)(.*)$/'
./step_definitions/pd.rb:42:in `/^I am a teacher who has just followed a workshop certificate link$/'
./support/hooks.rb:36:in `call'
./support/hooks.rb:36:in `block in '
features/pd/workshopCertificates.feature:7:in `Given I am a teacher who has just followed a workshop certificate link'
./step_definitions/pd.rb:89:in `create_enrollment'
./step_definitions/pd.rb:162:in `/^I create a workshop for course "([^"]*)" ([a-z]+) by "([^"]*)" with (\d+) (people|facilitators)(.*)$/'
./step_definitions/pd.rb:42:in `/^I am a teacher who has just followed a workshop certificate link$/'
./support/hooks.rb:36:in `call'
./support/hooks.rb:36:in `block in '
features/pd/workshopCertificates.feature:7:in `Given I am a teacher who has just followed a workshop certificate link'
./step_definitions/pd.rb:89:in `create_enrollment'
./step_definitions/pd.rb:162:in `/^I create a workshop for course "([^"]*)" ([a-z]+) by "([^"]*)" with (\d+) (people|facilitators)(.*)$/'
./step_definitions/pd.rb:42:in `/^I am a teacher who has just followed a workshop certificate link$/'
./support/hooks.rb:36:in `call'
./support/hooks.rb:36:in `block in '
features/pd/workshopCertificates.feature:7:in `Given I am a teacher who has just followed a workshop certificate link'
```

See https://cucumber-logs.s3.amazonaws.com/test/test/ChromeLatestWin7_pd_workshopCertificates_eyes_output.html?versionId=9KwLd3lA0Bxl1aAh5x8Q_Ms_1gwbUBXP

**Pull request merged**